src/MonoMod.RuntimeDetour/DetourManager.Managed.cs

```diff
@@ -360,6 +360,10 @@ internal sealed class ManagedDetourState {
 
                     UpdateEndOfChain();
                     UpdateChain(ilhook.Factory, out _);
+                } catch {
```

Comment:
The try range for this is a bit wide. An exception thrown before `UpdateChain` could cascade into another exception during the undoing process. 

In addition, I'm not sure that we want the `throw new InvalidOperationException("Trying to add an IL hook which was already added");` to cause the hook to be undone